pyproject.toml

```diff
@@ -134,3 +134,70 @@ markers = [
 
 [tool.black]
 line-length = 80
+
+[tool.isort]
+profile = "black"
+line_length = 79
+honor_noqa = true
+src_paths = ["awscli", "tests"]
+
+[tool.ruff]
+exclude = [
+    ".bzr",
+    ".direnv",
+    ".eggs",
+    ".git",
+    ".git-rewrite",
+    ".hg",
+    ".ipynb_checkpoints",
+    ".mypy_cache",
+    ".nox",
+    ".pants.d",
+    ".pyenv",
+    ".pytest_cache",
+    ".pytype",
+    ".ruff_cache",
+    ".svn",
+    ".tox",
+    ".venv",
+    ".vscode",
+    "__pypackages__",
+    "_build",
+    "buck-out",
+    "build",
+    "dist",
+    "node_modules",
+    "site-packages",
+    "venv",
+]
+
+# Format same as Black.
+line-length = 79
+indent-width = 4
+
+target-version = "py38"
```

Comment:
Is this still right? Are we waiting until botocore raises pins? I think the CLI v2 can be a little more aggressive.

Reply:
I assumed since our minimum version is Python 3.8, thats what we would have here. After a closer reading of the `ruff` docs, it would effectively be set to `py38` because we have this in our `pyproject.toml` too:

https://github.com/aws/aws-cli/blob/d4619a326382e52c9f0faeb29648df2b9a1489ed/pyproject.toml#L16

From the [docs](https://docs.astral.sh/ruff/settings/#target-version):

> If you're already using a pyproject.toml file, we recommend project.requires-python instead, as it's based on Python packaging standards, and will be respected by other tools. For example, Ruff treats the following as identical to target-version = "py38":
> 
>
> ```
> [project]
> requires-python = ">=3.8"
> ```

Reply:
Cool, we can probably yank this then and let `python_requires` be the single source of truth.